File: net/Link.Message.Client/content/complex/linear/FormFieldConfig.cs
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;

namespace Link.Message.Client.content.complex.linear
{

	using Guard = utils.Guard;

	/// <summary>
    /// Form类型线性消息内容项的表单配置
	/// </summary>
	public class FormFieldConfig
	{
		private int _width = 75;
		private Alignment _align = Alignment.Left;

		public FormFieldConfig()
		{

		}

		public FormFieldConfig(int width, Alignment align)
		{
			Guard.GuardReqiredObject(width, "width must be set value.");
			_width = width;
			_align = align;
		}

		public virtual int Width
		{
			get
			{
				return _width;
			}
			set
			{
				_width = value;
			}
		}

        [JsonConverter(typeof(StringEnumConverter))]
		public virtual Alignment Align
		{
            get
            {
                return _align;
            }
            set
            {
                _align = value;
            }
		}
	}

}

File: java/link-message-client/src/main/java/link/message/client/content/complex/linear/FormFieldConfig.java
package link.message.client.content.complex.linear;

import link.message.client.utils.Guard;

/**
 * Form类型线性消息内容项的表单配置
 * 
 * @author zhongt
 *
 */
public class FormFieldConfig {
	protected int width       = 75;
	protected Alignment align = Alignment.left;

	public FormFieldConfig() {
		
	}
	
	public FormFieldConfig(int width, Alignment align) {
		Guard.guardReqiredObject(width, "width must be set value.");
		this.width = width;
		this.align = align;
	}
	
	public int getWidth() {
		return width;
	}

	public void setWidth(int width) {
		this.width = width;
	}

	public Alignment getAlign() {
		return align;
	}

	public void setAlign(Alignment align) {
		this.align = align;
	}

}
